src/object/class_object.c

```diff
@@ -4112,50 +4116,53 @@ classobj_find_constraint_by_attrs (SM_CLASS_CONSTRAINT * cons_list, DB_CONSTRAIN
 	      len++;		/* increase name number */
 	    }
 
-	  if (!*attp && !*namep && !classobj_is_possible_constraint (cons->type, new_cons))
+	  if (*attp || *namep || classobj_is_possible_constraint (cons->type, new_cons))
 	    {
-	      for (i = 0; i < len; i++)
+	      continue;
+	    }
+
+	  for (i = 0; i < len; i++)
+	    {
+	      /* if not specified, ascending order */
+	      order = (asc_desc ? asc_desc[i] : 0);
+	      assert (order == 0 || order == 1);
+	      if (order != cons->asc_desc[i])
 		{
-		  /* if not specified, ascending order */
-		  order = (asc_desc ? asc_desc[i] : 0);
-		  assert (order == 0 || order == 1);
-		  if (order != cons->asc_desc[i])
-		    {
-		      break;	/* not match */
-		    }
+		  break;	/* not match */
 		}
+	    }
 
-	      if (i == len)
-		{
-		  if (filter_predicate)
-		    {
-		      if (!cons->filter_predicate)
-			{
-			  continue;
-			}
+	  if (i != len)
+	    {
+	      continue;
+	    }
 
-		      if (!filter_predicate->pred_string || !cons->filter_predicate->pred_string)
-			{
-			  continue;
-			}
+	  if (filter_predicate)
+	    {
```

Comment:
Checking `cons->filter_predicate` seems to be removed.

Reply:
Due to the conditions added to the 4105 line,
If filter_predicate is not NULL, cons->filter_predicate is also determined to be NULL.

`(filter_predicate && !cons->filter_predicate) || (!filter_predicate && cons->filter_predicate)`